daras_ai_v2/vector_search.py

```diff
@@ -823,7 +834,99 @@ def tabular_bytes_to_any_df(
+        slide_content = [f"Slide {slide_idx}:"]  # Markdown heading for slide
+        for shape in slide.shapes:
+            try:
+                if shape.has_text_frame:
+                    text = shape.text.strip()
+                    if text:
+                        slide_content.append(f"{text}")
+                if shape.has_table:
+                    slide_content.extend(pptx_format_table(shape.table))
+                if shape.has_chart:
+                    chart = shape.chart
+                    chart_title = (
+                        chart.chart_title.text_frame.text
+                        if chart.has_title
+                        else "Chart"
+                    )
+                    chart_text = [f"  {chart_title}:"]
+                    for series in chart.series:
+                        series_text = f"Series '{series.name}'"
+                        chart_text.append(series_text)
+                    slide_content.extend(chart_text)
+                if shape.shape_type == MSO_SHAPE_TYPE.GROUP:
+                    slide_content.extend(pptx_format_grouped_shape(shape))
+            except Exception as e:
+                # Catch any exceptions and append to slide content which should be handled better
+                slide_content.append(f"  Error processing shape: {e}")
+                
+        slides_text.append("\n".join(slide_content))
+    return slides_text
+
+
+def pptx_format_table(table) -> list[str]:
+    """
+    Formats a Shape-table into Markdown.
+    """
+    table_text = []
+    if len(table.rows) == 0:
+        return table_text
+
+    header_row = [cell.text.strip() for cell in table.rows[0].cells]
+    num_columns = len(header_row)
+
+    table_text.append(pptx_gen_table_row(header_row))
+    table_text.append(pptx_gen_table_row([":-:" for _ in header_row]))
+
+    for row_idx in range(1, len(table.rows)):
+        row = table.rows[row_idx]
+        row_text = [cell.text.strip() for cell in row.cells if cell.text.strip()]
+        if row_text:
+            table_text.append(pptx_gen_table_row(row_text))
+
+    return table_text
+
+
+def pptx_gen_table_row(row: list[str]) -> str:
+    """
+    Generates a formatted table row for Markdown.
+    """
+    return "| " + " | ".join([c.replace("\n", "<br />") for c in row]) + " |"
+
 
+def pptx_format_grouped_shape(group_shape) -> list[str]:
+    """
+    Recursively formats grouped shapes, extracting text from each.
+    """
+    group_text = []
+    check_recursively_for_text(group_shape, group_text)
+    return group_text
+
+def check_recursively_for_text(shape, group_text):
+    from pptx.enum.shapes import MSO_SHAPE_TYPE
+
+    if shape.shape_type == MSO_SHAPE_TYPE.GROUP:
+        for sub_shape in shape.shapes:
+            check_recursively_for_text(sub_shape, group_text)
+    else:
+        if hasattr(shape, "text"):
+            text = shape.text.strip()
+            if text:
+                group_text.append(f"{text}")
+                
```

Comment:
Referenced this StackOverflow post for implementing a recursive code block:  
https://stackoverflow.com/questions/51701626/how-to-extract-text-from-a-text-shape-within-a-group-shape-in-powerpoint-using